src/lib/token_list_gen.ts

```diff
@@ -345,3 +353,133 @@ export const arbListtoEtherscanList = (
   return list;
 };
 
+
+export const permitTest = async (pathOrUrl: string) => {
+  const l1TokenList = await getTokenListObj(pathOrUrl);
+  removeInvalidTokensFromList(l1TokenList);
+  const path = process.env.PWD +
+  '/src/ArbTokenLists/permit_' +
+  listNameToFileName(l1TokenList.name);
+  let prevPermitTokenList: ArbTokenList | undefined; 
+
+  if (existsSync(path)) {
+    const data = readFileSync(path)
+    console.log('Previous version of permit list found!');
+    
+    prevPermitTokenList = JSON.parse(data.toString()) as ArbTokenList;
+    isArbTokenList(prevPermitTokenList);
+  }
+
+  const newList = await generateTokenList(l1TokenList, prevPermitTokenList, {
+    includeUnbridgedL1Tokens: false,
+  });
+  const etherscanData = arbListtoEtherscanList(newList);
+  
+  const { l1 } = await getNetworkConfig();
+  const wallet = ethers.Wallet.createRandom().connect(l1.provider);
+  const spender = ethers.Wallet.createRandom().connect(l1.provider);
+  const value = ethers.utils.parseUnits("1.0", 18);
+  const deadline = ethers.constants.MaxUint256;
+  
+  const permitCalls = [];
+  let dictIdx = 0;
+  const idxToTokenInfo: { [key: number]: { l1Address: string; token: any} } = {};
+  const multicall = new ethers.Contract("0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696", multicallAbi, wallet);
```

Comment:
i tried using the multicaller from the sdk initially but it returns undefined instead of a bool for success, so I used the multicall contract to call tryAggregate